pkg/noderesourcetopology/least_numa.go

```diff
@@ -65,77 +87,69 @@ func leastNUMAPodScopeScore(pod *v1.Pod, zones topologyv1alpha2.ZoneList) (int64
 	identifier := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
 
 	resources := util.GetPodEffectiveRequest(pod)
+	// if a pod requests only non NUMA resources return max score
+	if onlyNonNUMAResources(nodes, resources) {
+		return framework.MaxNodeScore, nil
+	}
 
-	numaNodes := numaNodesRequired(identifier, qos, nodes, resources)
+	numaNodes, isMinAvgDistance := numaNodesRequired(identifier, qos, nodes, resources)
 	// pod's resources can't fit onto node, return MinNodeScore
 	if numaNodes == nil {
+		// score plugin should be running after resource filter plugin so we should always find sufficient amount of NUMA nodes
+		klog.Warningf("cannot calculate how many NUMA nodes are required for: %s", identifier)
 		return framework.MinNodeScore, nil
 	}
 
-	return normalizeScore(numaNodes.Count()), nil
+	return normalizeScore(numaNodes.Count(), isMinAvgDistance), nil
 }
 
-func normalizeScore(numaNodes int) int64 {
-	numaScore := framework.MaxNodeScore / highestNUMAID
-	return framework.MaxNodeScore - int64(numaNodes)*numaScore
+func normalizeScore(numaNodesCount int, isMinAvgDistance bool) int64 {
+	score := framework.MaxNodeScore - int64(numaNodesCount)*numaNodeScore
```

Comment:
*IF* `numaNodeScore` is used just here (seems the case?) maybe define it here for better locality, helping the reader

Reply:
done